src/content/app/genome-browser/state/track-panel/trackPanelSlice.ts

```diff
@@ -96,6 +103,25 @@ export const selectTrackPanelTab =
     );
   };
 
+export const updateTrackPanelTabForNewFocusObject = createAsyncThunk(
+  'track-panel-slice/updateTrackPanelTabWhenFocusObjectChanges',
+  (params: { genomeId: string; focusObjectId: string }) => {
+    const { genomeId, focusObjectId } = params;
+    const { type } = parseFocusObjectId(focusObjectId);
+    let trackPanelTab;
+    if (type === 'variant') {
+      trackPanelTab = TrackSet.VARIATION;
+    } else {
+      trackPanelTab = TrackSet.GENOMIC;
```

Comment:
Shouldn't there be a check for genomic?

Reply:
What we are doing here is one behaviour if focus object is a variant, and a different behaviour is focus object is anything else. No need to be more specific about focus object types.